<!DOCTYPE html>
<html lang="ja-JP">
<head>
<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-620926-3"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-620926-3');
</script>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="../style.css">
<link rel="stylesheet" href="../syntax-highlight.css">
<link rel="icon" type="image/png" href="../rurema.png">

<link rel="canonical" href="https://docs.ruby-lang.org/ja/latest/function/rb_rescue.html">

<title>function rb_rescue (Ruby 3.3 リファレンスマニュアル)</title>
<meta name="description" content="まず b_proc(data1) を実行し、その途中で例外が発生したら r_proc(data2) を実行します。
捕捉する例外は [[c:StandardError]] のサブクラスだけです。">
<script src="../script.js"></script>
</head>
<body>


<script type="application/ld+json">
{"@context":"http://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","item":"https://docs.ruby-lang.org/ja/latest/doc/index.html","name":"Ruby 3.3 リファレンスマニュアル","position":1},{"@type":"ListItem","item":"https://docs.ruby-lang.org/ja/latest/function/index.html","name":"関数一覧","position":2},{"@type":"ListItem","item":"https://docs.ruby-lang.org/ja/latest/function/rb_rescue.html","name":"rb_rescue","position":3}]}
</script>

<header>
  <nav>
    <ol class="inline-breadcrumb-list">
      <li>
        <a href="../doc/index.html">Ruby 3.3 リファレンスマニュアル</a>
      </li>
      <li>
        <a href="../function/index.html">関数一覧</a>
      </li>
      <li>
        rb_rescue 
      </li>
    </ol>
  </nav>

<h1>function rb_rescue</h1><span class="permalink">[<a href="https://github.com/rurema/doctree/edit/master/refm/capi/src/eval.c.rd#L676">edit</a>]</span>

</header>

<main>
<dl class="functionlist">
<dt>
  <code>VALUE rb_rescue(VALUE (*b_proc)(), VALUE data1, VALUE (*r_proc)(), VALUE data2)</code>
</dt>
<dd>
<dd class="function-description">
<p>
まず b_proc(data1) を実行し、その途中で例外が発生したら r_proc(data2) を実行します。捕捉する例外は <a href="../class/StandardError.html">StandardError</a> のサブクラスだけです。
</p>
</dd>

</dd>
</dl>
</main>

<footer id="footer">
  <a rel="license" href="https://creativecommons.org/licenses/by/3.0/">
    <img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/3.0/88x31.png" />
  </a>

  <a href="https://github.com/rurema/doctree/issues/new" id="feedback-link">フィードバックを送る</a>
  
    / <a href="https://github.com/rurema/doctree/edit/master/refm/capi/src/eval.c.rd#L676">このマニュアルを編集する</a>
  
  <script>if (window.URLSearchParams) { document.getElementById("feedback-link").search = new URLSearchParams({'body': document.location}); }</script>
</footer>
</body>
</html>
